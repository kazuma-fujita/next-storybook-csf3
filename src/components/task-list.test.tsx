import initStoryshots from "@storybook/addon-storyshots";
import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import * as stories from "./task-list.stories";

initStoryshots();

describe("TaskList", () => {
  const { WithPinnedTasks } = composeStories(stories);
  test("renders pinned tasks at the start of the list", () => {
    const pinnedTitle = WithPinnedTasks.args!.tasks![5].title; // "Task 6 (pinned)"
    render(<WithPinnedTasks />);
    const values = screen.getAllByDisplayValue(/^Task/);
    expect(values).toHaveLength(6);
    expect(screen.getByDisplayValue(pinnedTitle)).toBeInTheDocument();
    expect((values[0] as HTMLInputElement).value).toBe(pinnedTitle);
  });
});
