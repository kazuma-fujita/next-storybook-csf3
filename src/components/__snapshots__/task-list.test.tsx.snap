// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Sign In Form Default 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="wrap"
  >
    <div
      className="content"
    >
      <input
        name="mailAddress"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="メールアドレス"
        type="text"
      />
    </div>
    <div
      className="content"
    >
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="パスワード"
        type="password"
      />
    </div>
    <div
      className="content"
    >
      <input
        type="submit"
        value="ログイン"
      />
    </div>
  </div>
</form>
`;

exports[`Storyshots Sign In Form Empty Error 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="wrap"
  >
    <div
      className="content"
    >
      <input
        name="mailAddress"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="メールアドレス"
        type="text"
      />
    </div>
    <div
      className="content"
    >
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="パスワード"
        type="password"
      />
    </div>
    <div
      className="content"
    >
      <input
        type="submit"
        value="ログイン"
      />
    </div>
  </div>
</form>
`;

exports[`Storyshots Sign In Form Filled 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="wrap"
  >
    <div
      className="content"
    >
      <input
        name="mailAddress"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="メールアドレス"
        type="text"
      />
    </div>
    <div
      className="content"
    >
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="パスワード"
        type="password"
      />
    </div>
    <div
      className="content"
    >
      <input
        type="submit"
        value="ログイン"
      />
    </div>
  </div>
</form>
`;

exports[`Storyshots Sign In Form Filled Success 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="wrap"
  >
    <div
      className="content"
    >
      <input
        name="mailAddress"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="メールアドレス"
        type="text"
      />
    </div>
    <div
      className="content"
    >
      <input
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="パスワード"
        type="password"
      />
    </div>
    <div
      className="content"
    >
      <input
        type="submit"
        value="ログイン"
      />
    </div>
  </div>
</form>
`;

exports[`Storyshots Task Item Archived 1`] = `
<div
  className="list-item TASK_ARCHIVED"
>
  <label
    className="checkbox"
  >
    <input
      defaultChecked={true}
      disabled={true}
      name="checked"
      type="checkbox"
    />
    <span
      className="checkbox-custom"
      onClick={[Function]}
    />
  </label>
  <div
    className="title"
  >
    <input
      placeholder="Input title"
      readOnly={true}
      type="text"
      value="OK Test Task"
    />
  </div>
  <div
    className="actions"
    onClick={[Function]}
  />
</div>
`;

exports[`Storyshots Task Item Default 1`] = `
<div
  className="list-item TASK_INBOX"
>
  <label
    className="checkbox"
  >
    <input
      defaultChecked={false}
      disabled={true}
      name="checked"
      type="checkbox"
    />
    <span
      className="checkbox-custom"
      onClick={[Function]}
    />
  </label>
  <div
    className="title"
  >
    <input
      placeholder="Input title"
      readOnly={true}
      type="text"
      value="OK Test Task"
    />
  </div>
  <div
    className="actions"
    onClick={[Function]}
  >
    <a
      onClick={[Function]}
    >
      <span
        className="icon-star"
      />
    </a>
  </div>
</div>
`;

exports[`Storyshots Task Item Pinned 1`] = `
<div
  className="list-item TASK_PINNED"
>
  <label
    className="checkbox"
  >
    <input
      defaultChecked={false}
      disabled={true}
      name="checked"
      type="checkbox"
    />
    <span
      className="checkbox-custom"
      onClick={[Function]}
    />
  </label>
  <div
    className="title"
  >
    <input
      placeholder="Input title"
      readOnly={true}
      type="text"
      value="OK Test Task"
    />
  </div>
  <div
    className="actions"
    onClick={[Function]}
  >
    <a
      onClick={[Function]}
    >
      <span
        className="icon-star"
      />
    </a>
  </div>
</div>
`;

exports[`Storyshots Task List Default 1`] = `
<div
  className="list-items"
>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 1"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 2"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 3"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 4"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 5"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 6"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`Storyshots Task List Empty 1`] = `
<div
  className="list-items"
>
  <div
    className="wrapper-message"
  >
    <span
      className="icon-check"
    />
    <div
      className="title-message"
    >
      You have no tasks
    </div>
    <div
      className="subtitle-message"
    >
      Sit back and relax
    </div>
  </div>
</div>
`;

exports[`Storyshots Task List Loading 1`] = `
<div
  className="list-items"
>
  <div
    className="loading-item"
  >
    <span
      className="glow-checkbox"
    />
    <span
      className="glow-text"
    >
      <span>
        Loading
      </span>
       
      <span>
        cool
      </span>
       
      <span>
        state
      </span>
    </span>
  </div>
  <div
    className="loading-item"
  >
    <span
      className="glow-checkbox"
    />
    <span
      className="glow-text"
    >
      <span>
        Loading
      </span>
       
      <span>
        cool
      </span>
       
      <span>
        state
      </span>
    </span>
  </div>
  <div
    className="loading-item"
  >
    <span
      className="glow-checkbox"
    />
    <span
      className="glow-text"
    >
      <span>
        Loading
      </span>
       
      <span>
        cool
      </span>
       
      <span>
        state
      </span>
    </span>
  </div>
  <div
    className="loading-item"
  >
    <span
      className="glow-checkbox"
    />
    <span
      className="glow-text"
    >
      <span>
        Loading
      </span>
       
      <span>
        cool
      </span>
       
      <span>
        state
      </span>
    </span>
  </div>
  <div
    className="loading-item"
  >
    <span
      className="glow-checkbox"
    />
    <span
      className="glow-text"
    >
      <span>
        Loading
      </span>
       
      <span>
        cool
      </span>
       
      <span>
        state
      </span>
    </span>
  </div>
  <div
    className="loading-item"
  >
    <span
      className="glow-checkbox"
    />
    <span
      className="glow-text"
    >
      <span>
        Loading
      </span>
       
      <span>
        cool
      </span>
       
      <span>
        state
      </span>
    </span>
  </div>
</div>
`;

exports[`Storyshots Task List With Pinned Tasks 1`] = `
<div
  className="list-items"
>
  <div
    className="list-item TASK_PINNED"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 6 (pinned)"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 1"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 2"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 3"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 4"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
  <div
    className="list-item TASK_INBOX"
  >
    <label
      className="checkbox"
    >
      <input
        defaultChecked={false}
        disabled={true}
        name="checked"
        type="checkbox"
      />
      <span
        className="checkbox-custom"
        onClick={[Function]}
      />
    </label>
    <div
      className="title"
    >
      <input
        placeholder="Input title"
        readOnly={true}
        type="text"
        value="Task 5"
      />
    </div>
    <div
      className="actions"
      onClick={[Function]}
    >
      <a
        onClick={[Function]}
      >
        <span
          className="icon-star"
        />
      </a>
    </div>
  </div>
</div>
`;
